types/shared.ts

```diff
@@ -1285,53 +1285,6 @@ export type Camelize<T> = {
     : never;
 };
 
-// export type Optionalize<T> = T extends object ?
-//   & {
-//     [K in KeysWithUndefined<T>]?: Optionalize<T[K]>;
-//   }
-//   & {
-//     [K in Exclude<keyof T, KeysWithUndefined<T>>]: Optionalize<T[K]>;
-//   }
-//   : T;
-
-// export type KeysWithUndefined<T> = {
-//   [K in keyof T]-?: (undefined | null) extends T[K] ? K : never;
-// }[keyof T];
-
-// export type Optionalize<T> = (
-//   & {
-//     [K in KeysWithUndefined<T>]?: Optionalize<T[K]>;
-//   }
-//   & {
-//     [K in Exclude<keyof T, KeysWithUndefined<T>>]: (
-//       // deno-lint-ignore ban-types
-//       T[K] extends object ? Object extends Pick<T[K], keyof T[K]> ? T[K] : Optionalize<T[K]> : T[K]
-//     );
-//   }
-// );
-
-// export type KeysWithUndefined<T> = {
-//   [K in keyof T]-?: undefined extends T[K] ? K
-//     : null extends T[K] ? K
-//     : never;
-// }[keyof T];
-
-// export type Optionalize<T> = T extends object ? (
-//   & {
-//     [K in KeysWithUndefined<T>]?: T[K] extends Collection<any, any> ? T[K]
-//       : T[K] extends any[] ? T[K]
-//       : Optionalize<T[K]>;
-//   }
-//   & {
-//     [K in Exclude<keyof T, KeysWithUndefined<T>>]: T[K] extends object ? Object extends Pick<T[K], keyof T[K]> ? T[K]
-//     : T[K] extends Collection<any, any> ? T[K]
-//     : T[K] extends any[] ? T[K]
-//     : Optionalize<T[K]>
-//       : T[K];
-//   }
-// )
-//   : T;
-
```

Comment:
tell me if I should move this to another PR

Reply:
I think this is the answer 🙃